apps/nextjs/hooks/useCompleteLogin.ts

```diff
@@ -0,0 +1,39 @@
+import Router from "next/router";
+import { useEffect } from "react";
+
+import { AuthCompleteResponse } from "pages/api/auth/complete";
+import { getJsonPostSettings } from "utils/fetch";
+import { OAuthManager } from "utils/oauth";
+
+/**
+ * Complete OAuth authentication flow and redirect user to frontpage.
+ */
+export const useCompleteLogin = (
+  provider: string,
```

Comment:
Can we / should we explicitly declare the valid providers with typescript as string literals? I'd expect it to be tied to the OAuth initiators in some fashion if yes, but unsure if it's worth the hassle at this point since it'd mean implementing validation that already exists on the backend.

Reply:
As noted in elsewhere, this'd seem like a good idea at least on the state storage interface.

Reply:
Done.